src/main/java/com/upserve/uppend/lookup/LongLookup.java

```diff
@@ -253,17 +253,21 @@ public void flush() {
             cacheCopy = new ConcurrentHashMap<>(writeCache);
         }
         ArrayList<Future> futures = new ArrayList<>();
-        cacheCopy.forEach((path, data) ->
-            futures.add(AutoFlusher.flushExecPool.submit(() -> {
-                try {
-                    log.trace("cache flushing {}", path);
-                    data.flush();
-                    log.trace("cache flushed {}", path);
-                } catch (Exception e) {
-                    log.error("unable to flush " + path, e);
+        cacheCopy.forEach(
+                (path, data) ->
+                {
+                    if (data.isDirty())
```

Comment:
This seems like something data should decide from within `flush`

Reply:
Seems like a lot of overhead to submit it as a task just to check the whether you need to flush.
Being able to trim (formerly known as purge) the write cache certainly is the best way to reduce the flushes, but the auto flusher pool gets pretty busy.